components/merkledb/src/proof_list_index/proof.rs

```diff
@@ -58,17 +58,75 @@ impl ProofOfAbsence {
 
 /// An enum that represents a proof of existence for a proof list elements.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum ListProof<V> {
+pub enum ProofVariant<V> {
     /// A branch of proof in which both children contain requested elements.
-    Full(Box<ListProof<V>>, Box<ListProof<V>>),
+    Full(Box<ProofVariant<V>>, Box<ProofVariant<V>>),
     /// A branch of proof in which only the left child contains requested elements.
-    Left(Box<ListProof<V>>, Option<Hash>),
+    Left(Box<ProofVariant<V>>, Option<Hash>),
     /// A branch of proof in which only the right child contains requested elements.
-    Right(Hash, Box<ListProof<V>>),
+    Right(Hash, Box<ProofVariant<V>>),
     /// A leaf of the proof with the requested element.
     Leaf(V),
     /// Proof of absence of an element with the specified index.
-    Absent(ProofOfAbsence),
+    Absent(Hash),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ListProof<V> {
+    length: u64,
+    proof: ProofVariant<V>,
+}
+
+impl<V> ListProof<V>
+where
+    V: BinaryValue,
+{
+    pub fn new(length: u64, proof: ProofVariant<V>) -> Self {
```

Comment:
Is `pub(crate)` visibility not enough in this case ?

Reply:
I wonder if one want to construct synthetic proofs to check API endpoints.